app/models/user.rb

```diff
@@ -1,48 +1,48 @@
+# frozen_string_literal: true
+
 class User < ApplicationRecord
-	attr_accessor :remember_token
- 	has_many :posts, dependent: :destroy
- 	before_save :downcase_email
-	validates :name,  presence: true, length: { maximum: 50 }
-  	VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
-  	validates :email, presence: true, length: { maximum: 255 },
+  attr_accessor :remember_token
+  has_many :posts, dependent: :destroy
+  before_save :downcase_email
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 255 },
                     format: { with: VALID_EMAIL_REGEX },
                     uniqueness: { case_sensitive: false }
 
-    has_secure_password
-    validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
-	
-	def downcase_email
-      self.email.downcase
-    end 
-
-
-	class << self
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
 
-	      def digest(string)
-	        cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
-	                                                      BCrypt::Engine.cost
-	        BCrypt::Password.create(string, cost: cost)
-	      end
+  def downcase_email
+    email.downcase
+  end
 
-	      def new_token
-	        SecureRandom.urlsafe_base64
-	      end
-	  end
+  class << self
+    def digest(string)
+      cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                    BCrypt::Engine.cost
+      BCrypt::Password.create(string, cost: cost)
+    end
 
-	 def remember
-      self.remember_token = User.new_token
-      update_attribute(:remember_digest, User.digest(remember_token)) # creates a digest of the remember me token before saving in the database
+    def new_token
+      SecureRandom.urlsafe_base64
     end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
```

Comment:
This should be called before a user is created using the before_create action

Reply:
ok